src/crypto/Address.dto.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import {
+  Address,
+  ByronAddress,
+  BaseAddress,
+  PointerAddress,
+  EnterpriseAddress,
+  RewardAddress,
+} from '@emurgo/react-native-haskell-shelley'
+import {Logger} from '../utils/logging'
+
+// NOTE: ** IMPORTANT ** The order matters
+const ADDRESS_TYPES = Object.freeze([
```

Comment:
Heads up: `react-native-haskell-shelley` is disabled in flow, so you can't rely on types (everything is just `any`). You can enable it but flow's report will be polluted by many errors unrelated to your code, which come from the fact that `react-native-haskell-shelley` does not cover the whole API given by `cardano-serialization-lib`.

Reply:
Roger that, removing the typing, setting as `any`.